Enable shadowrealm testing for abort api

rwaldron opened the pull request.

rwaldron committed:
Enable shadowrealm testing for abort api

annevk commented:
Let's block on this as per https://github.com/web-platform-tests/wpt/pull/41985.

ptomato committed:
Split AbortSignal.timeout() tests into separate files

AbortSignal.timeout() isn't exposed in ShadowRealm environments, so it
shouldn't be tested there. Split it out into a separate file, and add
another file for ShadowRealm environments testing that the property is not
present.

ptomato commented:
Rebased and added some additional changes since `AbortSignal.timeout()` isn't exposed in ShadowRealm.

ptomato commented on dom/abort/AbortSignal.any.js:
```diff
@@ -10,31 +12,3 @@ async_test(t => {
   s.onabort = t.unreached_func("abort event handler called");
   t.step_timeout(() => { t.done(); }, 2000);
```
FWIW, `step_timeout()` is implemented with `setTimeout()` which also doesn't work in ShadowRealm. We could move this one to a Window/Worker-only file as well.

rwaldron replied:
You should feel entirely free to drive these PRs to completion, as I trust your expertise. 

From a person opinion point: seems unfortunate that setTimeout isn't exposed in a browser ShadowRealm :(

rwaldron replied:
WDYT about adding in a fallback "setTimeout" implementation here? https://github.com/web-platform-tests/wpt/blob/master/tools/serve/serve.py#L426-L440

In Test262 we provide a setTimeout for use in Atomics testing: 

```js
  if (this.setTimeout === undefined) {
    (function(that) {
      that.setTimeout = function(callback, delay) {
        let p = Promise.resolve();
        let start = Date.now();
        let end = start + delay;
        function check() {
          if ((end - Date.now()) > 0) {
            p.then(check);
          }
          else {
            callback();
          }
        }
        p.then(check);
      }
    })(this);
  }
```

If you'd rather not, I will support your decision!

ptomato replied:
Ah, I like this idea a lot. I was thinking of sending a callable through the realm boundary that, when called, would make the incubating realm terminate the test after the timeout expired, but I wasn't sure how to go about that. This approach is much simpler, thanks.

ptomato replied:
See https://github.com/web-platform-tests/wpt/pull/43639 - the `setTimeout` is included there. I put it in a separate PR because hopefully those test harness improvements don't need to block on the HTML spec.

ptomato commented:
Was going to try to merge this at the same time as #41985, but looks like that one needs some more discussion. So I'll go ahead and merge this one. Thanks for the review.
